scripts/test.sh

```diff
@@ -2,33 +2,17 @@
 
 thread_count=$(getconf _NPROCESSORS_ONLN)
 
-if which flutter >/dev/null; then
-    flutter pub global activate junitreport
-    flutter test --concurrency=$thread_count --coverage
-    TEST_CODE=$?
-    
-    # coverage
-    flutter pub global activate remove_from_coverage
-    flutter pub global run remove_from_coverage:remove_from_coverage -f coverage/lcov.info -r '\.g\.dart$'
+if [ -n "$NO_OLM" ]; then 
+    tagFlag="-x"
 else
-    dart test --concurrency=$thread_count --coverage=coverage
-    TEST_CODE=$?
-    
-    # coverage -> broken see https://github.com/dart-lang/test/issues/1698
-    dart pub global activate coverage
-
-    #reporton="--report-on=lib/"
-    if [ -n "$NO_OLM" ]; then reporton="--report-on=lib/src --report-on=lib/msc_extensions"; fi
-
-    dart pub global run coverage:format_coverage -i coverage/  --lcov -o coverage/lcov.info $reporton
-    dart pub global activate remove_from_coverage
-    dart pub global run remove_from_coverage:remove_from_coverage -f coverage/lcov.info -r '\.g\.dart$'
+    tagFlag="-t"
 fi
 
-# coverage html report
-genhtml -o coverage coverage/lcov.info || true
 
-# https://github.com/eriwen/lcov-to-cobertura-xml
-lcov_cobertura.py coverage/lcov.info || true
+dart test --concurrency=$thread_count --coverage=$1 $tagFlag 'olm'
 
-exit $TEST_CODE
+# lets you do more stuff like reporton
+dart pub global activate coverage
+dart pub global run coverage:format_coverage --lcov -i $1 -o $1/lcov.info --report-on=lib/
+dart pub global activate remove_from_coverage
+dart pub global run remove_from_coverage:remove_from_coverage -f $1/lcov.info -r '\.g\.dart$'
```

Comment:
You should either check, that $1 is actually set to a valid directory or so, or give it some default value. I don't think we should write to /lcov.info or even worse, if you have a space in it allow injecting arbitrary parameters?